pkg/hsskey/hsskey.go

```diff
@@ -165,14 +167,34 @@ func GetAllHss(warnings io.Writer, verboseDangerous bool, eepromPattern string,
 		}
 	}
 
+	// Attempt to get HSS from files.
 	if hssFiles != "" {
-		hssFileArr := strings.Split(hssFiles, ",")
-		hss, err := GetHssFromFile(warnings, verboseDangerous, hssFileArr, 0)
+		fi, err := os.Stat(hssFiles)
 		if err != nil {
-			io.WriteString(warnings, fmt.Sprintf("Error parsing file %s for HSS: %v\n", hss, err))
-		}
-		if len(hss) > 0 {
-			hssList = append(hssList, hss...)
+			io.WriteString(warnings, fmt.Sprintf("Failed to stat HSS files: %v\n", err))
+		} else {
+			hssFileArr := []string{}
+
+			// Check if hssFiles is a directory or a file. Add all regular files in the base directory.
+			switch mode := fi.Mode(); {
+			case mode.IsDir():
+				filepath.Walk(hssFiles, func(fpath string, info os.FileInfo, _ error) error {
+					if info.Mode().IsRegular() {
+						hssFileArr = append(hssFileArr, fpath)
+					}
+					return nil
+				})
+			case mode.IsRegular():
+				hssFileArr = append(hssFileArr, hssFiles)
+			}
+
+			hss, err := GetHssFromFile(warnings, verboseDangerous, hssFileArr, 0)
+			if err != nil {
```

Comment:
I don't see the need for this test.
hssList = append(hssList, hss...) 
works fine whether hss is 0 length or not
https://go.dev/play/p/0j69OpB9Ax8

Reply:
True, but I'd prefer to have the printing done here rather than suppress the error or force the underlying function to print. The append is always done anyways (not an else block), but I'd prefer to keep the check on the err print.